packages/langium/src/grammar/langium-grammar.langium

```diff
@@ -74,6 +74,12 @@ ParserRule:
     (definesHiddenTokens?='hidden' '(' (hiddenTokens+=[AbstractRule:ID] (',' hiddenTokens+=[AbstractRule:ID])*)? ')')? ':'
     definition=Alternatives ';';
 
+InfixRule: 'infix' RuleNameAndParams 'on' call=RuleCall ':' operators=InfixRuleOperators ';';
+
+InfixRuleOperators: precedences+=InfixRuleOperatorList ('>' precedences+=InfixRuleOperatorList)*;
+
+InfixRuleOperatorList: operators+=Keyword ('|' operators+=Keyword)*;
```

Comment:
I assume the associativity is left-to-right?

Shall we add an option to change that?

```
InfixRuleOperatorList:
    (associativity=Associativity 'assoc')?
    operators+=Keyword ('|' operators+=Keyword)*;

Associativity returns string: 'left' | 'right';
```

If omitted, the default should be `left`. With this, you can realize assignments as operators:

```
infix BinaryExpression on PrimaryExpression:
    '*' | '/'
    > '+' | '-'
    > right assoc '=' | '*=' | '/=' | '+=' | '-=';
```

Maybe also allow `no assoc` to forbid more than two operands on the same level?